frame/nomination-pools/src/lib.rs

```diff
@@ -1973,6 +2030,29 @@ pub mod pallet {
 }
 
 impl<T: Config> Pallet<T> {
+	/// Slightly more customizable version of [`frame_system::ensure_signed_or_root`].
```

Comment:
Pretty useful function, I can see it being useful in other pallets too if it were promoted to frame_system